Create new blog entry: 2022-01-13-ci-cd-definition-how-to-apply-it-in-your-organization-gologic.md (#30)
xNok/xNok: website/content/english/blogs/2022-01-13-ci-cd-definition-how-to-apply-it-in-your-organization-gologic.md

---

> title: "CI/CD Definition: How to Apply It in Your Organization?"

Author: ```suggestion
title: "[What is CI/CD and how to apply it in your organization?"
```

date: 2022-01-13
draft: false
categories: ["DevOps"]
tags: ["CI/CD", "DevOps"]
images:

>   - https://prod-files-secure.s3.us-west-2.amazonaws.com/bd16a693-0437-45a1-9aec-255351a830a8/d8292f84-dba1-454a-8577-b0b2a1ba2b3a/Article_Blogue_CICD_Application_Organisation.png

Author: ```suggestion
  - https://www.gologic.ca/wp-content/uploads/2022/01/Article_Blogue_CICD_Application_Organisation.png.webp
```

## Summary

This article defines **CI/CD** as core practices for a successful **DevOps** transformation. It explains the difference between Continuous Delivery and Continuous Deployment, highlighting Continuous Deployment as the ultimate goal where code changes are automatically validated and deployed. The article also emphasizes the importance of Continuous Testing, observability, and monitoring. To prepare your organization, you need to define the proper infrastructure for collaboration between “Dev” and “Ops”, and implement a fail-fast approach.

## Key Concepts

*   **Continuous Integration (CI):** A development practice where developers regularly merge code changes into a central repository, after which automated builds and tests are run.
*   **Continuous Delivery (CD):** An extension of continuous integration to make sure that you can release new changes to your customers quickly in a sustainable way.
*   **Continuous Deployment:** The holy grail of DevOps, where every change that passes the automated tests is automatically deployed to production.
*   **Fail-Fast Approach:** Implementing validation at every step of the process so teams can quickly identify and fix issues.

> ## Infrastructure
> 
> *   Micro-services
> *   Resources
> *   Shared services
> *   Platform
> *   Infrastructure

Author: ```suggestion
```
